modules/planning/scenarios/side_pass/side_pass_scenario.cc

```diff
@@ -193,138 +203,46 @@ bool SidePassScenario::IsFarFromIntersection(const Frame& frame) {
-        adc_sl_boundary.end_s()) {  // such vehicles are behind the ego car.
-      ADEBUG << " - It is behind ADC.";
-      continue;
-    }
-
-    // check s distance
-    constexpr double kAdcDistanceThreshold = 15.0;  // unit: m
-    if (obstacle->PerceptionSLBoundary().start_s() >
-        adc_sl_boundary.end_s() + kAdcDistanceThreshold) {
-      // vehicles are far away
-      ADEBUG << " - It is too far ahead.";
-      continue;
-    }
-
-    if (adc_sl_boundary.end_s() +
-            side_pass_context_.scenario_config_.min_front_obstacle_distance() >
-        obstacle->PerceptionSLBoundary().start_s()) {
-      // front obstacle is too close to side pass
-      ADEBUG << " - It is too close to side-pass.";
-      continue;
-    }
-
-    // check driving_width
-    if (!IsBlockingDrivingPathObstacle(reference_line, obstacle)) {
-      ADEBUG << " - It is not blocking our way.";
-      continue;
-    }
-
-    bool is_blocked_by_others = false;
-    if (side_pass_context_.scenario_config_.enable_obstacle_blocked_check() &&
-        !IsParked(reference_line, obstacle)) {
-      for (const auto* other_obstacle : path_decision.obstacles().Items()) {
-        if (other_obstacle->Id() == obstacle->Id()) {
-          continue;
-        }
-        if (other_obstacle->IsVirtual()) {
-          continue;
-        }
-        if (other_obstacle->PerceptionSLBoundary().start_l() >
-                obstacle->PerceptionSLBoundary().end_l() ||
-            other_obstacle->PerceptionSLBoundary().end_l() <
-                obstacle->PerceptionSLBoundary().start_l()) {
-          // not blocking the backside vehicle
-          continue;
-        }
-
-        double delta_s = other_obstacle->PerceptionSLBoundary().start_s() -
-                         obstacle->PerceptionSLBoundary().end_s();
-        if (delta_s < 0.0 || delta_s > kAdcDistanceThreshold) {
-          continue;
-        } else {
-          // TODO(All): fixed the segmentation bug for large vehicles, otherwise
-          // the follow line will be problematic.
-          is_blocked_by_others = true;
-          break;
-        }
+    double distance_between_adc_and_obstacle = 0.0;
+    if (IsBlockingObstacleToSidePass(
+            frame, obstacle,
+            side_pass_context_.scenario_config_.block_obstacle_min_speed(),
+            side_pass_context_.scenario_config_.min_front_obstacle_distance(),
+            side_pass_context_.scenario_config_.enable_obstacle_blocked_check(),
+            &distance_between_adc_and_obstacle)) {
+      exists_a_blocking_obstacle = true;
+      if (distance_to_closest_blocking_obstacle < 0 ||
+          distance_between_adc_and_obstacle <
+              distance_to_closest_blocking_obstacle) {
+        // Only record the closest front blocking obstacle.
+        distance_to_closest_blocking_obstacle =
+            distance_between_adc_and_obstacle;
+        front_blocking_obstacle_id_ = obstacle->Id() + "_0";
```

Comment:
Modifying the obstacle id is dangerous as multiple places are using id to match them.

Reply:
Before this refactor, the obstacle id was modified like this. I wasn't quite sure why obstacle's id should be appended by a "_0"; therefore, I retained this convention.